depinject/container.go

```diff
@@ -431,6 +453,10 @@ func (c container) formatResolveStack() string {
 	return buf.String()
 }
 
+func (c *container) addPreference(p preference) {
+	c.preferences = append(c.preferences, p)
```

Comment:
This is where you want to build a map of in type name to preference. No need to build a list